std-bits/table/src/main/java/org/enso/table/data/column/operation/BinaryOperationBoolean.java

```diff
@@ -0,0 +1,105 @@
+package org.enso.table.data.column.operation;
+
+import org.enso.table.data.column.builder.BuilderForBoolean;
+import org.enso.table.data.column.storage.BoolStorage;
+import org.enso.table.data.column.storage.ColumnBooleanStorage;
+import org.enso.table.data.column.storage.ColumnStorage;
+import org.enso.table.data.column.storage.type.BooleanType;
+import org.enso.table.data.column.storage.type.NullType;
+import org.enso.table.data.column.storage.type.StorageType;
+import org.enso.table.problems.BlackholeProblemAggregator;
+
+/** Binary operation for (Boolean, Boolean) => Boolean. */
+public abstract class BinaryOperationBoolean extends BinaryOperationBase<Boolean> {
+  private final boolean preserveNulls;
+
+  public BinaryOperationBoolean() {
+    this(true);
+  }
+
+  public BinaryOperationBoolean(boolean preserveNulls) {
+    super(BooleanType.INSTANCE);
+    this.preserveNulls = preserveNulls;
+  }
+
+  @Override
+  public ColumnStorage<Boolean> applyMap(ColumnStorage<?> left, Object rightValue) {
```

Comment:
Given that the standard way to provide a more efficient implementation is by overriding `applyMapOverBoolStorage`, then maybe `applyMap` should become `final`?

Same for zip.